src/config.rs

```diff
@@ -329,6 +388,21 @@ impl Config {
             // If everything fails, give up
             bail!("Could not determine user cache directory");
         };
+        let tls_backend = match raw_config.updates.tls_backend {
+            #[cfg(feature = "native-tls")]
+            RawTlsBackend::NativeTls => TlsBackend::NativeTls,
+            #[cfg(feature = "rustls")]
+            RawTlsBackend::Rustls => TlsBackend::Rustls,
+            #[cfg(feature = "rustls-with-native-roots")]
+            RawTlsBackend::RustlsWithNativeRoots => TlsBackend::RustlsWithNativeRoots,
+            // when compiling without all TLS backend features, we want to handle config error.
+            #[allow(unreachable_patterns)]
+            _ => return Err(anyhow!(
+                    "Unsupported TLS backend: {}. This tealdeer build has support for the following options: {}",
+                 raw_config.updates.tls_backend,
+                 SUPPORTED_TLS_BACKENDS.map(|b| b.to_string()).join(", ")
+                ))
```

Comment:
Very nice error message! Let's add a test that checks the output if this case occurs. To do so, you can use for example `let testenv = TestEnv::new().no_default_features().with_feature("rustls");` in our integration tests at `tests/lib.rs`.

I discovered that our implementation of `TestEnv::command` is not correct at the moment, sorry about that. You should change line 123 of `tests/lib.rs` from
```rust
            build = build.arg(format!("--feature {}", self.features.join(",")));
```
to
```rust
            build = build.arg("--features").arg(self.features.join(","));
```